incubator/hnc/pkg/controllers/config_controller.go

```diff
@@ -42,13 +56,24 @@ func (r *ConfigReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	// achieve the reconciliation.
 	r.Log.Info("Reconciling cluster-wide HNC configuration.")
 
+	// TODO: This is only for debuging and testing in the early development stage. Remove this when the
+	// implementation is completed.
+	r.logSpec(inst)
+
+	// Create cooresponding ObjectReconcilers for newly added types, if needed.
+	// TODO: Returning an error here will make controller-runtime to requeue the object to be reconciled again.
+	// A better way to handle this is to display the error as part of the HNCConfig.status field to suggest the error is a
+	// permanent problem and to stop controller-runtime from retrying.
+	if err := r.createObjectReconcilers(inst); err != nil {
```

Comment:
Maybe add a `TODO` for something like `syncObjectReconcilers()`? It will also include removing obsolete object reconcilers besides `createObjectReconcilers()`.

Reply:
Good point. Added the TODO for renaming in future. Feel free to take a look.